lib/ecto/query/builder/join.ex

```diff
@@ -62,15 +62,15 @@ defmodule Ecto.Query.Builder.Join do
   end
 
   def escape({:__aliases__, _, _} = module, _vars, _env) do
-    {:_, {nil, module}, nil, %{}}
+    {:_, quote(do: Ecto.Queryable.to_query(unquote(module))), nil, %{}}
```

Comment:
Same thing here. It is probably best to do it on the planner and not when the query is being built.